flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/bridging/BridgingSqlAggFunction.java

```diff
@@ -117,6 +116,22 @@ public static BridgingSqlAggFunction of(
                 dataTypeFactory, typeFactory, kind, resolvedFunction, typeInference);
     }
 
+    /** Creates an instance of a aggregate function during translation. */
+    public static BridgingSqlAggFunction of(
+            FlinkContext context,
+            FlinkTypeFactory typeFactory,
+            ContextResolvedFunction resolvedFunction) {
+        final DataTypeFactory dataTypeFactory = context.getCatalogManager().getDataTypeFactory();
```

Comment:
That's only personal preference, but since we just hava a couple of chained calls, why not inline them in the `of()` call instead of assigning them to vars?

Reply:
Yes, this might be personal preference. Local variable help in breaking a problem into subproblems. I find this makes the code much more readable and no IDE shows you a warning for this. Not everything that can be inlined should be inlined if you can give it a nice variable name.